### DISCUSSION TITLE
encryption/decryption support? openssl is a big library

### DISCUSSION THREAD

**Discussion by camelcigarettes** [mpv-player/mpv#13908] [Q&A]
you can use api calls in the openssl libraries to encrypt and decrypt audio video (not sure how muxing mixes in with filters, maybe optimizations through some kind of wrapper or function macro for ffmpeg?) 

anyways, the commands I use for basic encryption on the command line is

openssl enc -aes128 -in paddingbear.mp4 -out padding.bear.emp4 -iter 2048

and to decrypt its
openssl enc -d -aes128 -in paddingbear.emp4 -iter 2048

this puts a basic aes128 password on the file for decryption for the openssl binary or standardized libraries to decrypt the file

and for mpv you can do something like, mpve -encrypt paddingtonbear.mp4 and it'll output paddingbear.emp4 , and playing would be as simple as mpv paddingtonbear.emp4

it may take some crunching on cpu time but it could save some time when it comes to stuff like computer security

**Comment by kasper93** [comment]
You can specify `encryption_key` and `decryption_key` for mp4 in libavformat. This should accomplish probably what you want.
